optimum/onnxruntime/modeling_seq2seq.py

```diff
@@ -596,22 +606,28 @@ def forward(
             encoder_outputs = self.encoder(input_ids=input_ids, attention_mask=attention_mask)
 
         # Decode
-        if past_key_values is None or self.decoder_with_past is None:
-            decoder_outputs = self.decoder(
-                input_ids=decoder_input_ids,
-                encoder_hidden_states=encoder_outputs.last_hidden_state,
-                encoder_attention_mask=attention_mask,
-            )
+        decoder_inputs = {
+            "input_ids": decoder_input_ids
+            if (past_key_values is None or self.decoder_with_past is None)
+            else decoder_input_ids[:, -1:],  # Cut decoder_input_ids if past is used
+            "encoder_hidden_states": encoder_outputs.last_hidden_state,
+            "encoder_attention_mask": attention_mask,
+        }
+        if past_key_values is not None and self.decoder_with_past:
+            decoder_inputs["past_key_values"] = past_key_values
+        if labels is not None:
+            decoder_inputs["labels"] = labels
+        decoder_outputs = self.decoder(**decoder_inputs)
+
+        if labels is None:
+            return Seq2SeqLMOutput(logits=decoder_outputs.logits, past_key_values=decoder_outputs.past_key_values)
         else:
-            decoder_outputs = self.decoder_with_past(
-                input_ids=decoder_input_ids[:, -1:],  # Cut decoder_input_ids if past is used
-                past_key_values=past_key_values,
-                encoder_hidden_states=encoder_outputs.last_hidden_state,
-                encoder_attention_mask=attention_mask,
```

Comment:
`self.decoder_with_past` should not be removed and should be used when `use_cache` is True and when `past_key_values` have been computed

Reply:
Yes, you ar right. Putting it back!